modules/oe_translation_poetry/tests/Functional/PoetryMockTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_translation_poetry\Functional;
+
+use Drupal\oe_translation_poetry_mock\PoetryMock;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the poetry mock.
+ */
+class PoetryMockTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'system',
+    'content_translation',
+    'tmgmt',
+    'oe_translation',
+    'oe_translation_poetry',
+    'oe_translation_poetry_mock',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->container->get('config.factory')->getEditable('oe_translation_poetry.settings')
+      ->set('service_wsdl', PoetryMock::getWsdlUrl())
+      ->save();
+  }
+
+  /**
+   * Tests that requests to Poetry are being mocked.
+   */
+  public function testRequestResponse(): void {
+    /** @var \Drupal\oe_translation_poetry\Poetry $poetry */
+    $poetry = $this->container->get('oe_translation_poetry.client');
+
+    // Assert that the Poetry service gets properly instantiated.
+    $expected_settings = [];
+    $expected_settings['identifier.code'] = 'WEB';
+    $expected_settings['identifier.sequence'] = 'EWCMS_SEQUENCE';
+    $expected_settings['identifier.year'] = date('Y');
+    $expected_settings['service.wsdl'] = PoetryMock::getWsdlUrl();
+    $expected_settings['service.username'] = 'admin';
+    $expected_settings['service.password'] = 'admin';
+    $expected_settings['notification.username'] = 'admin';
+    $expected_settings['notification.password'] = 'admin';
+
+    $settings = $poetry->getSettings();
+    foreach ($expected_settings as $name => $setting) {
+      $this->assertEqual($setting, $settings[$name]);
+    }
+
+    // Create a test message and check that Mocked responses are returned.
+    /** @var \EC\Poetry\Messages\Requests\CreateTranslationRequest $message */
+    $message = $poetry->get('request.create_translation_request');
+
+    $test_id = 1;
+    $id = $message->getIdentifier();
+    $id->setNumber('40012');
```

Comment:
Usually on production we only set the numero once we have received one from poetry, on the very first request you should send a sequence identifier and receive the numero from poetry.